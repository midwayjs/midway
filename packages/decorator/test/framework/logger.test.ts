import {
  Logger,
  getClassMetadata,
  CLASS_KEY_CONSTRUCTOR,
  LOGGER_KEY,
} from '../../src';

class Test {
  constructor(@Logger('aaa') aaa: any) {
    // ignore
  }

  @Logger()
  logger: any;

  @Logger('bbb')
  bbb: any;
}

describe('/test/framework/logger.test.ts', () => {
  it('logger decorator should be ok', () => {
    let data = getClassMetadata(CLASS_KEY_CONSTRUCTOR, Test);
    expect(data).toStrictEqual({
      0: {
        key: 'aaa',
        type: LOGGER_KEY,
      },
    });

    data = getClassMetadata(LOGGER_KEY, Test);
    expect(data).toStrictEqual([
      { key: 'logger', propertyName: 'logger' },
      { key: 'bbb', propertyName: 'bbb' },
    ]);
  });
});
