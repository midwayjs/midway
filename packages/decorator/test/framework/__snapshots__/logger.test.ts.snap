// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/test/framework/logger.test.ts logger decorator should be ok 1`] = `
Object {
  "bbb": Object {
    "impl": true,
    "key": "logger",
    "metadata": Object {
      "identifier": "bbb",
    },
    "propertyName": "bbb",
  },
  "logger": Object {
    "impl": true,
    "key": "logger",
    "metadata": Object {
      "identifier": undefined,
    },
    "propertyName": "logger",
  },
}
`;
